pkg/webhook/board_update.go

```diff
@@ -0,0 +1,142 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-resty/resty"
+	"github.com/google/go-github/github"
+	"github.com/syndesisio/pure-bot/pkg/config"
+	"go.uber.org/zap"
+	"regexp"
+	"strconv"
+)
+
+type boardUpdate struct{}
+
+func (h *boardUpdate) EventTypesHandled() []string {
+	return []string{"issues", "pull_request"}
+}
+
+type column struct {
+	name string
+	id   string
+}
+
+var stateMapping = map[string]column{}
+
+var zenHubApi = "https://api.zenhub.io"
+
+func (h *boardUpdate) HandleEvent(eventObject interface{}, gh *github.Client, config config.RepoConfig, logger *zap.Logger) error {
+
+	// initialise from config if needed
+	if len(stateMapping) == 0 {
+		for _, col := range config.Board.Columns {
+			c := column{col.Name, col.Id}
+			for _, event := range col.Events {
+				logger.Info("Mapping " + event + " to " + col.Name)
+				stateMapping[event] = c
+			}
+		}
+	}
+
+	switch event := eventObject.(type) {
+	case *github.IssuesEvent:
+		return h.handleIssuesEvent(event, gh, config, logger)
+	case *github.PullRequestEvent:
+		return h.handlePullRequestEvent(event, gh, config, logger)
+	default:
+		return nil
+	}
+}
+
+func (h *boardUpdate) handleIssuesEvent(event *github.IssuesEvent, gh *github.Client, config config.RepoConfig, logger *zap.Logger) error {
+
+	var messageType = "issues"
+
+	number := strconv.Itoa(*event.Issue.Number)
+	logger.Debug("Handling issuesEvent for issue " + number)
+	logger.Debug("Issue Action: " + *event.Action)
+
+	eventKey := messageType + "_" + *event.Action
+	col, ok := stateMapping[eventKey]
+	if ok {
+		return moveIssueOnBoard(config, number, col)
+	} else {
+		logger.Debug("Ignore ummapped event: " + eventKey)
+	}
+
+	return nil
+}
+
+func (h *boardUpdate) handlePullRequestEvent(event *github.PullRequestEvent, gh *github.Client, config config.RepoConfig, logger *zap.Logger) error {
+
+	var messageType = "pull_request"
+
+	prNumber := strconv.Itoa(*event.PullRequest.Number)
+	logger.Info("Handling pullReqestEvent for PR " + prNumber)
+	logger.Info("PR Action: " + *event.Action)
+
+	commits, _, err := gh.PullRequests.ListCommits(context.Background(), event.Repo.Owner.GetLogin(), event.Repo.GetName(),
+		*event.PullRequest.Number, nil)
+
+	if err != nil {
+		logger.Error("Failed to retrieve commits")
+		return nil
+	}
+
+	for _, commit := range commits {
+
+		message := *commit.Commit.Message
+		logger.Debug("Processing commit message: " + message)
+
+		// get issue id from commit message
+		var re = regexp.MustCompile(`(?mi)^.*((Fix)(.*)(#{1})([0-9]+))+.*$`)
```

Comment:
I think this regexp is a bit too lose, e.g. it would also match a comment like "I'm not sure whether this really fixes the actual issue. However I'm pretty sure it's related to #1234, too".

I.e. it should match on whitespace e.g. like in (fix[^\s]*\s+#[0-9]+). Not sure why you us that many grouping parentheses and why `#{1}` instead of `#` (or `\#` if it needs to be escaped).

Reply:
it's because I am miserable at regex